### PR TITLE
Add Order to OnAppStart

### PR DESCRIPTION
**Pull request by vahid-sohrabloo**
Add Order To onAppStart.
With This change we can order to run function in runStartupHooks.

### CODE REVIEW

**Comment by vahid-sohrabloo** [comment]
@anonx  I added test. Why you add needs testting tag?

**Comment by anonx** [comment]
@vahid-sohrabloo, thank you for this PR. Everything is OK, "needs testing" is a label for @revel/core to make sure everything is correct and to double check things.

**Comment by vahid-sohrabloo** [comment]
@anonx Thank you

File: server.go [hunk 1]
```diff
@@ -131,12 +132,32 @@ func Run(port int) {
...
 	for _, hook := range startupHooks {
-		hook()
+		hook.f()
 	}
 }
 
-var startupHooks []func()
+type StartupHook struct {
+	order int
+	f     func()
```
Comment (brendensoares) [inline server.go]: Not sure I like such a vague variable name on a `struct`. Thoughts?

**Comment by brendensoares** [comment]
Overall, good enhancement.

File: server.go [hunk 2]
```diff
@@ -174,6 +195,10 @@ var startupHooks []func()
 // This can be useful when you need to establish connections to databases or third-party services,
 // setup app components, compile assets, or any thing you need to do between starting Revel and accepting connections.
 //
-func OnAppStart(f func()) {
-	startupHooks = append(startupHooks, f)
+func OnAppStart(f func(), order ...int) {
+	o := 1
```
Comment (notzippy) [inline server.go]: I think the default value should be 5 and we have some sort of guideline as to the range of values to be used.
ie  Default / normal is 5, high / first is 1, low / last is 10. 
So people know that revel builtin startuphooks will run at 5, if they want something before or after it should be higher or lower.

Comment (brendensoares) [inline server.go]: Agreed.

Comment (vahid-sohrabloo) [inline server.go]: I'm not Agree. people can use negative numbers.

File: server.go (hunk 1 above)
Comment (vahid-sohrabloo) [inline server.go]: maybe. do you have name suggest?

Comment (brendensoares) [inline server.go]: `hook.callback()` seems appropriate.

File: server.go (hunk 2 above)
Comment (brendensoares) [inline server.go]: That's true. We can assume all hooks will run at the same level just as they do currently and can be changed to different priority if needed. Should be good for backward compatibility.

**Pull request merged**